app/experimenter/experiments/tests/test_models.py

```diff
@@ -1155,21 +1156,56 @@ def test_rollouts_monitoring_dashboard_returns_correct_formatted_url(self):
             ),
         )
 
-    def test_rollouts_monitoring_dashboard_returns_url_when_rollout_is_complete(self):
+    @parameterized.expand(
+        [
+            (
+                False,
+                NimbusExperiment.Status.DRAFT,
+                NimbusExperimentFactory.Lifecycles.CREATED,
+            ),
+            (
+                True,
+                NimbusExperiment.Status.COMPLETE,
+                NimbusExperimentFactory.Lifecycles.ENDING_APPROVE_APPROVE,
+            ),
+            (
+                True,
+                NimbusExperiment.Status.LIVE,
+                NimbusExperimentFactory.Lifecycles.LIVE_ENROLLING,
+            ),
+            (
+                True,
+                NimbusExperiment.Status.LIVE,
+                NimbusExperimentFactory.Lifecycles.LAUNCH_APPROVE_APPROVE,
+            ),
+            (
+                False,
+                NimbusExperiment.Status.PREVIEW,
+                NimbusExperimentFactory.Lifecycles.PREVIEW,
+            ),
+        ]
+    )
+    def test_rollouts_monitoring_dashboard_returns_url(
+        self, valid_status, status, lifecycle
+    ):
         experiment = NimbusExperimentFactory.create_with_lifecycle(
-            NimbusExperimentFactory.Lifecycles.ENDING_APPROVE,
+            lifecycle=lifecycle,
             slug="rollout-1-slug",
             is_rollout=True,
-            status=NimbusExperiment.Status.COMPLETE,
-        )
-        expected_slug = "rollout_1_slug"
-        self.assertEqual(
-            experiment.rollout_monitoring_dashboard_url,
-            settings.ROLLOUT_MONITORING_URL.format(
-                slug=expected_slug,
-            ),
+            status=status,
```

Comment:
You don't need to explicitly set the status here, that's part of the lifecycles.

Reply:
🤯